src/sentry/issues/endpoints/organization_searches.py

```diff
@@ -110,6 +110,7 @@ def post(self, request: Request, organization: Organization) -> Response:
             sort=result["sort"],
             visibility=result["visibility"],
         )
+        assert saved_search.type is not None
```

Comment:
wouldn't it be better to check result["type"] in the validator?

Reply:
it is checked there -- but the model can do some defaulting on `type` so I have to check it here as well

technically the database schema is wrong (it should be `NOT NULL` for that column) but fixing that is a lot more involved